replica/table.cc

```diff
@@ -642,24 +642,26 @@ table::seal_active_memtable(flush_permit&& flush_permit) noexcept {
                 tlogger.warn("Memtable flush failed due to: {}. Dropped due to shutdown", e);
                 co_return;
             }
+            auto abort_on_error = [e] () {
+                // At this point we don't know what has happened and it's better to potentially
+                // take the node down and rely on commitlog to replay.
+                //
+                // FIXME: enter maintenance mode when available.
+                // since replaying the commitlog with a corrupt mutation
+                // may end up in an infinite crash loop.
+                on_internal_error_noexcept(tlogger, format("Memtable flush failed due to: {}", e));
+
+                // If the node is configured to not abort on internal error,
+                // but propagate it up the chain, we can't do anything reasonable
+                // at this point.
+                abort();
```

Comment:
Wow, this on_internal_error_noexcept is really confusing. Why does it even exist in this form? I'll open a separate issue about that. https://github.com/scylladb/seastar/issues/1095